### PR TITLE
enhance the empty state by using typography theme

### PR DESCRIPTION
**Pull request by codershona**
#  Enhance the empty state for the hooks view #3044

**Explanation of my method:**
I had tried to customize the theme by adding a component and to expect the behaviour I had to try added an updated hook method as when we try to get updated hooks pf contained in empty search.

### CODE REVIEW

File: ui/src/views/Hooks/ListHooks/index.jsx [hunk 1]
```diff
@@ -35,6 +35,10 @@ export default class ListHooks extends Component {
     this.props.history.push('/hooks/create');
   };
 
+  handleHookUpdateSearch = () => {
+    this.props.history.push('/hooks/update');
```
Comment (djmitche) [inline ui/src/views/Hooks/ListHooks/index.jsx:39]: What is this route for?

File: ui/src/views/Hooks/ListHooks/index.jsx [hunk 2]
```diff
@@ -89,7 +93,11 @@ export default class ListHooks extends Component {
             softSearch
             tree={tree}
             onEmptySearch={
-              <Typography variant="subtitle1">
+              <Typography
+                variant="subtitle1"
+                placeholder="Updated Hook contains"
```
Comment (djmitche) [inline ui/src/views/Hooks/ListHooks/index.jsx:98]: What does this mean?  I believe when this is displayed, there are no hooks, so nothing has been "updated"..

File: ui/src/views/Hooks/ListHooks/index.jsx [hunk 3]
```diff
@@ -89,7 +93,11 @@ export default class ListHooks extends Component {
...
+                color="inherit"
```
Comment (djmitche) [inline ui/src/views/Hooks/ListHooks/index.jsx:99]: Why is this required?

File: ui/src/views/Hooks/ListHooks/index.jsx [hunk 4]
```diff
@@ -89,7 +93,11 @@ export default class ListHooks extends Component {
...
+                onClick={this.handleHookUpdateSearch}>
                 No items for search term {hookSearch}
```
Comment (djmitche) [inline ui/src/views/Hooks/ListHooks/index.jsx:101]: This text looks different from what #3044 shows. A bit of exploring the git history shows this issue already got fixed, in #2779.  So, maybe there's nothing to change here?

Comment (codershona) [reply ui/src/views/Hooks/ListHooks/index.jsx:101]: So where I am a bit that where could I change as I thought it was about to enhance the empty state by using the color component in the typography theme.

Comment (djmitche) [reply ui/src/views/Hooks/ListHooks/index.jsx:101]: Nope, the expected behavior doesn't say anything about color:
>  If there are no hooks to display, it should say something other than "No items for search term".

Comment (djmitche) [reply ui/src/views/Hooks/ListHooks/index.jsx:101]: Oh, I've misunderstood the issue, too!

I'll update the issue itself with a better explanation.

Comment (codershona) [reply ui/src/views/Hooks/ListHooks/index.jsx:101]: ok.

File: ui/src/views/Hooks/ListHooks/index.jsx [hunk 5]
```diff
@@ -89,8 +89,8 @@ export default class ListHooks extends Component {
             softSearch
             tree={tree}
             onEmptySearch={
-              <Typography variant="subtitle1">
-                No items for search term {hookSearch}
+              <Typography variant="h3" component="h3" color="primary">
+                No hooks are defined {hookSearch}
```
Comment (djmitche) [inline ui/src/views/Hooks/ListHooks/index.jsx:93]: This is looking better, but now when there *is* a search term, for example `proj-taskcluster`, it's going to say something like
> No hooks are defined proj-taskcluster

which doesn't make a lot of sense.  How could it be made to say 

> No hooks are defined

when there's no hooks defined and

> No hooks found matching "proj-taskcluster"

When there's a search term but no matching results?

**Comment by djmitche** [comment]
We're all back to work now so I'll have a look at this soon.  Sorry for the delay!

File: ui/src/views/Hooks/ListHooks/index.jsx [hunk 6]
```diff
@@ -37,23 +37,29 @@ export default class ListHooks extends Component {
...
+    if (query.search !== hookSearch) {
+      const newQuery = {
         ...query,
         search: hookSearch,
-      }),
-    });
+
+      };
+      history.push({
+        search: stringify(newQuery, {addQueryPrefix: true}),
```
Comment (djmitche) [inline ui/src/views/Hooks/ListHooks/index.jsx:49]: I'm curious why this was required?  This causes `stringify` to add a `?` to the beginning of the query, but [history already does that](https://github.com/ReactTraining/history/blob/3f69f9e07b0a739419704cffc3b3563133281548/modules/LocationUtils.js#L18-L23).

It looks like history is documented as expecting the `?` to be present so this is not a bad change to make, but doesn't address the issue this PR is solving.

File: ui/src/views/Hooks/ListHooks/index.jsx (hunk 5 above)
Comment (djmitche) [reply ui/src/views/Hooks/ListHooks/index.jsx:99]: It looks like this still only ever says "No hooks are defined" -- I don't see the other strings anywhere in the PR.

**Review by djmitche — COMMENTED** [review]
Good progress here :)

If you run `yarn lint --fix, it will fix the stylistic issues and then the checks on the PR will make a test version of the site, which we can use to check out your changes.  As it is, because of the lint issues, those checks are failing.

File: ui/src/views/Hooks/ListHooks/index.jsx (hunk 5 above)
Comment (codershona) [reply ui/src/views/Hooks/ListHooks/index.jsx:99]: Hello, @djmitche  I am a bit confused will the string will look like this, 

`No hooks are defined ${hooksearch}`

File: ui/src/views/Hooks/ListHooks/index.jsx (hunk 6 above)
Comment (codershona) [reply ui/src/views/Hooks/ListHooks/index.jsx:49]: Here if i try to use this **logic** will it would be correct to use it,

```
handleHookSearchSubmit = hookSearch => {
    const query = parse(window.location.search.slice(1));
    const {
        location,
        history
    } = this.props;

    if (query.search !== hookSearch) {
        const newQuery = {
            ...query,
            search: hookSearch
        }
        else {
            location.search = '?' + location.search;
        }

    }
}
```

Comment (djmitche) [reply ui/src/views/Hooks/ListHooks/index.jsx:49]: I think it would be correct to omit the change entirely in this PR, and focus on fixing the issue in #3044.

File: ui/src/views/Hooks/ListHooks/index.jsx (hunk 5 above)
Comment (djmitche) [reply ui/src/views/Hooks/ListHooks/index.jsx:99]: That would have the same effect as what's already written.

It might be good to make a diagram or notes of what the different states are, and what the component should look like in each of those states.  Then think of a way to put the component in each of those states in your browser, so you can verify whether your changes have achieved the desired result.

File: ui/src/views/Hooks/ListHooks/index.jsx [hunk 7]
```diff
@@ -91,12 +85,12 @@ export default class ListHooks extends Component {
...
-            searchTerm={hookSearch}
+            searchTerm={hookSearch || null}
             softSearch
             tree={tree}
             onEmptySearch={
-              <Typography variant="h3" color="secondary">
-                No hooks are defined {hookSearch}
+              <Typography variant="subtitle1">
+                 {`No hooks are defined ${hookSearch}`}
               </Typography>
```
Comment (codershona) [inline ui/src/views/Hooks/ListHooks/index.jsx:94]: I had added a string logic here but I am still confused about the no display hooks logics.

**Comment by djmitche** [comment]
Let's take a bit of a step back here.  What is the issue looking for?

* When _____ then the component should ______
* When _____ then the component should ______

Once you answer that, we can take the next step.

**Comment by djmitche** [comment]
I'm going to close this PR -- we can continue this conversation on #3304.
